File: backend/src/main/java/com/example/demo/model/Todo.java
package com.example.demo.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;
import java.time.Instant;
import java.time.LocalDate;

@Entity
@Data
public class Todo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id; 
    
    private Long userId; // New: To link the todo to a user
    
    private String title;
    private String description;
    private LocalDate dueDate; 
    private String priority;
    private String status; 
    private String category;
    private Instant createdAt;
    private Instant updatedAt;
    private Integer orderIndex;

    public Todo() {
        this.createdAt = Instant.now();
        this.updatedAt = Instant.now();
        this.status = "active";
        this.priority = "medium";
    }
}

File: src/types/todo.ts
export interface Todo {
  id: string;
  title: string;
  description?: string;
  due_date?: string;
  priority: 'low' | 'medium' | 'high';
  status: 'active' | 'completed';
  category?: string;
  created_at: string;
  updated_at: string;
  user_id?: string;
  order_index: number;
}

export interface TodoFilter {
  status: 'all' | 'active' | 'completed';
  priority?: 'low' | 'medium' | 'high';
  category?: string;
  search: string;
}

export interface TodoSort {
  field: 'created_at' | 'due_date' | 'priority' | 'title';
  direction: 'asc' | 'desc';
}

export interface User {
  id: string;
  username: string;
  email?: string;
  created_at: string;
  preferences: {
    theme: 'light' | 'dark';
    notifications: boolean;
  };
}